### PR TITLE
Blog - Building a Monorepo with Rust

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
@adamgordonbell [Preview link](https://659688702f611e5fbda9d4a1--ephemeral-narwhal-3ac3b2.netlify.app/blog/rust-monorepo/)

[Airtable link](https://airtable.com/appbtxxY2KaPio7dV/tblsY8AOw3qPBIUdZ/viwBflqOY7OlQqiNg/recSIIWrtbhxA4ecR?blocks=hide)

File: blog/_posts/2024-01-11-rust-monorepo.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+### A Simpler Solution: Earthly
+
+Now that you know how to run and manage a monorepo locally, it's time to configure a build tool that can containerize the packages and run tests in a simpler way. [Earthly](https://earthly.dev/) is a great alternative for these tasks.
+
+Earthly is a build tool that enables each service or library to independently handle its own build and test cycles. It also supports caching, which means that only the packages that have been updated will retrigger their build.
+
+To configure Earthly in this project, you need to add an `Earthfile` in each of the packages and one parent `Earthfile` at the root of the project. Here's the `Earthfile` for the `filter` library:
+
+~~~{.earthfile caption="Earthfile"}
```
Comment (adamgordonbell) [inline blog/_posts/2024-01-11-rust-monorepo.md:256]:
```suggestion
~~~{.dockerfile caption="Earthfile"}
```

Comment (adamgordonbell) [reply blog/_posts/2024-01-11-rust-monorepo.md:256]: The blog doesn't know Earthly, but docker is closest

File: blog/_posts/2024-01-11-rust-monorepo.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+# test runs the tests present in the package
+test: 
+    FROM +build
+    RUN cargo test
+~~~
+
+This file enables the package to be built into a self-contained artifact, which you can then reference easily in other services. Here's the `Earthfile` for the `ingest` service:
+
+~~~{.earthfile caption="Earthfile"}
```
Comment (adamgordonbell) [inline blog/_posts/2024-01-11-rust-monorepo.md:281]:
```suggestion
~~~{.dockerfile caption="Earthfile"}
```

File: blog/_posts/2024-01-11-rust-monorepo.md [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    FROM +build
+    ENTRYPOINT ["./ingest"]
+    SAVE IMAGE ingest:latest
+~~~
+
+This file defines its own build and Docker steps and makes use of the artifact generated by the `filter` library to include it in the final binary.
+
+The `Earthfile` for the `cleanup` service is quite similar to that of `ingest`:
+
+~~~{.earthfile caption="Earthfile"}
```
Comment (adamgordonbell) [inline blog/_posts/2024-01-11-rust-monorepo.md:321]:
```suggestion
~~~{.dockerfile caption="Earthfile"}
```

File: blog/_posts/2024-01-11-rust-monorepo.md [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+# docker creates docker image cleanup:latest
+docker:
+    FROM +build
+    ENTRYPOINT ["./cleanup"]
+    SAVE IMAGE cleanup:latest
+~~~
+
+Finally, here's the parent `Earthfile` that allows you to use the steps defined in the package `Earthfile`s easily:
+
+~~~{.earthfile caption="Earthfile"}
```
Comment (adamgordonbell) [inline blog/_posts/2024-01-11-rust-monorepo.md:359]:
```suggestion
~~~{.dockerfile caption="Earthfile"}
```

**Review by adamgordonbell — CHANGES_REQUESTED** [review]
A couple small changes, otherwise good.

**Comment by msali123** [comment]
@adamgordonbell [Updated link](https://659815bd6247541408c2e0e6--ephemeral-narwhal-3ac3b2.netlify.app/blog/rust-monorepo/)

**Pull request merged**